src/code/z_parameter.c

```diff
@@ -3094,26 +3091,24 @@ void Interface_Draw(GlobalContext* globalCtx) {
                         interfaceCtx->counterDigits[3] -= 10;
                     }
 
-                    phi_s2 = 42;
-                    if (interfaceCtx->counterDigits[2] != 0) {
-                        OVERLAY_DISP = Gfx_TextureI8(OVERLAY_DISP, D_02003040[interfaceCtx->counterDigits[2]], 8, 16,
-                                                     phi_s2, 190, 8, 16, 1024, 1024);
-                        phi_s2 = 50;
+                    svar3 = 42;
+
+                    if (interfaceCtx->counterDigits[2]) {
+                        OVERLAY_DISP = Gfx_TextureIA8(
+                            OVERLAY_DISP, ((u8*)gHUDCounterDigit0Tex + (8 * 16 * interfaceCtx->counterDigits[2])), 8,
```

Comment:
Changes like this doesn't make sense at all.
The old `D_02003040[]` was an array of textures, so doing this kind of trickery seems bad. We should extract those assets as the proper texture array it is.

Reply:
its not an array of textures, and it won't match the same way if you try that.

Reply:
How do you know it isn't?

Reply:
try changing these to not use a u8 cast and the pointer math and it will not be the same asm. This is more than likely what they wrote, and there isn't really much we can do about it besides making macros to accommodate.

Reply:
Also if I don't do it like this the entire bss section breaks.

Reply:
I mean, an array of pointers of `u8`
```
u8 D_02003040[10][0x80] = {
    { /* First texture */ },
    { /* Second texture */ },
    { /* Third texture */ },
    // ...
};
```

Reply:
yes i know, and that affects matching somehow, and even breaks the bss section, which is why there were the random bss dummy variables in this file before.

Reply:
So many things affect bss reordering, that we cant really say that doing it this way "fixes it". Even something like changing the amount of includes in the file will change what dummy variables are required to match. The fact that changing this happened to fix bss is a coincidence more than anything

Reply:
its literally not, again try messing with it before saying im wrong

Reply:
I agree with you that in this case it fixes bss, but it will be messed up again in the future when other things change. There are many many variables to bss reordering

Reply:
well this issue should be able to be resolved then?

Reply:
not sure what you mean.
I am just making sure its clear that bss reodering being fixed does not mean it proves that this is the correct way to write this. its going to be messed up again after something else seemingly unrelated changes, cause its very volatile.

If there are other matching issues that this fixes then thats fine, but bss reodering should not be considered as proof

Reply:
i've done quite a bit to this file and nothing else has messed it up, i could include tons of headers with symbols in them and it does not do anything. the only thing affecting this is it not being a 2d array